lib/matplotlib/pyplot.py

```diff
@@ -2358,6 +2401,9 @@ def _autogen_docstring(base):
 # to determine if they should trigger a draw.
 install_repl_displayhook()
 
+# Set up the backend.
+switch_backend(rcParams["backend"])
```

Comment:
No it doesn't, because the QApplication isn't created until a figure is created.  Indeed, you snippet works as expected.
Sure, the snippet would fail with ImportError if PyQt5 is not installed, but that seems to be asking a bit too much (just wrap it in a try... except yourself); in fact I think a fast fail is less confusing.

Reply:
I am apparently confused about the details of various hooks get installed or not.... :sheep:

Reply:
In general nothing should happen until you create a widget; we then start the loop lazily.